src/runtime/classbase.cs

```diff
@@ -354,6 +354,9 @@ public static IntPtr tp_repr(IntPtr ob)
         public static void tp_dealloc(IntPtr ob)
         {
             ManagedType self = GetManagedObject(ob);
+            if (self == null)
```

Comment:
Yes (https://stackoverflow.com/questions/24468667/whats-the-difference-between-tp-clear-tp-dealloc-and-tp-free#24863227). This is one of my attempts to fix the reference counting issues. I can either remove this from this PR and include it in a new one, or leave it in as it can definitely not hurt.